lapis/scheduler.py

```diff
@@ -101,23 +101,23 @@ def _schedule_job(self, job) -> Drone:
         for cluster in self.drone_cluster:
             drone = cluster[0]
             cost = 0
-            resource_types = {*drone.resources.keys(), *job.resources.keys()}
-            for resource_type in resource_types:
-                if resource_type not in drone.resources.keys():
-                    cost = float("Inf")
-                    break
-                elif resource_type not in job.resources:
-                    cost += drone.pool_resources[resource_type] \
-                        - drone.resources[resource_type]
-                elif (drone.pool_resources[resource_type]
-                      - drone.resources[resource_type]) < job.resources[resource_type]:
+            resources = drone.theoretical_available_resources
+            for resource_type in job.resources:
+                if resources.get(resource_type, 0) < job.resources[resource_type]:
+                    # Inf for all job resources that a drone does not support
+                    # and all resources that are too small to even be considered
                     cost = float("Inf")
                     break
                 else:
-                    cost += (drone.pool_resources[resource_type]
-                             - drone.resources[resource_type]) // \
-                        job.resources[resource_type]
-            cost /= len(resource_types)
+                    try:
+                        cost += 1 / (resources.get(resource_type, 0)
```

Comment:
Doesn't a default of 0 cause a ``ZeroDivisionError``? You may want to have no default and fail faster with a ``KeyError``.

Reply:
Thanks, took care on this.